### PR TITLE
[feature] #4390: include trigger id in trigger entrypoint

### PR DESCRIPTION
**Pull request by mversic**
## Description

* Added `SetKeyValue`/`RemoveKeyValue` for trigger
* added corresponding metadata changed events for trigger
* added trigger id to trigger entrypoint arguments
* verified with a simple test

### Linked issue

Closes #{issue_number} 

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs [hunk 1]
```diff
@@ -10,29 +10,43 @@ extern crate alloc;
...
 #[iroha_smart_contract::main]
 fn main(owner: AccountId) {
-    let asset_cursor = FindAllAssets
-        .fetch_size(FetchSize::new(Some(NonZeroU32::try_from(1).dbg_unwrap())))
-        .execute()
-        .dbg_unwrap();
-
-    let (_batch, cursor) = asset_cursor.into_parts();
+    // NOTE: Fields of QueryOutputCursor are private so we guess the layout by encoding/decoding
+    let asset_cursor = QueryOutputCursor::decode_all(
```
Comment (DCNick3) [inline client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs:36]: This seems like a hack.. Is this something our users are supposed to to?

**Review by DCNick3 — DISMISSED** [review]

File: client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs (hunk 1 above)
Comment (mversic) [reply client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs:36]: no, ofc not. This is not something for users to do. This is only for this test to verify that query batching works without exposing internals of the cursor in the API. It's a dirty hack for sure

Comment (DCNick3) [reply client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs:36]: Maybe add a comment for clarity? Integration tests can be used as examples, but some things should not be taken as an example :sweat_smile:

Comment (mversic) [reply client/tests/integration/smartcontracts/query_assets_and_save_cursor/src/lib.rs:38]: done

**Review by Erigara — APPROVED** [review]
Please notify SDKs that permission token name was changed.

**Review by DCNick3 — APPROVED** [review]

**Pull request merged**